lib/Optimizer/CodeGen/LowerToQIR.cpp

```diff
@@ -365,12 +365,11 @@ class OneTargetRewrite : public ConvertOpToLLVMPattern<OP> {
     auto context = parentModule->getContext();
     std::string qirQisPrefix(cudaq::opt::QIRQISPrefix);
     std::string instName = instOp->getName().stripDialect().str();
-    if (!instOp->template hasTrait<cudaq::Hermitian>() && instOp.getIsAdj())
-      instName += "dg";
 
     if (numControls == 0) {
       // There are no control bits, so call the function directly.
-      auto qirFunctionName = qirQisPrefix + instName;
+      auto qirFunctionName =
+          qirQisPrefix + instName + (instOp.getIsAdj() ? "__adj" : "");
```

Comment:
I was wondering if this was going to add the suffix to all of the ops, but then I remember I fixed the ops so that adjoint can only be added to those that are not self-adjoint.